crates/ide/src/highlight_related.rs

```diff
@@ -520,6 +516,12 @@ pub(crate) fn highlight_yield_points(
                     if block_expr.async_token().is_none() {
                         continue;
                     }
+
+                    // Async blocks act similar to closures. So we want to
+                    // highlight their exit points too.
+                    let exit_points = hl_exit_points(sema, block_expr.async_token(), block_expr.clone().into());
+                    merge_map(&mut res, exit_points);
```

Comment:
Hmm, so the thing is, when the cursor is on `async` we only want to render the yield points, not exit points (this is useful for functions where you can get either depending on whether you put the cursor on the `async` or `fn` keyword). For async blocks you don't really have a simple way to get the exit points though so having it render both on `async` does make sense imo (but not when the cursor is on an `await). 


So I would want to branch this by `block_expr.async_token() == token` to only trigger this if the cursor is on the `async` token of the block

Reply:
Sorry, it took a bit, but it should be good now.

Reply:
No worries